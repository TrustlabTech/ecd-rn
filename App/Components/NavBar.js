import React, { Component } from 'react'
import {
    Text,
    View,
    StyleSheet,
    Platform,
    TouchableHighlight
} from 'react-native'

import Config from '../Config'
import Routes from '../Routes'
import SystemBar from './SystemBar'

export default class NavBar extends Component {

  constructor(props) {
    super(props)
    this.state = {
      route: props.route,
      navigator: props.navigator
    }
  }


  canGoBack() {
    return this.props.navigator.getCurrentRoutes().length > 1 ?
      true : false
  }

  render() {

    var leftButton
    if( this.props.leftButtonText &&
        this.props.leftButtonAction){
      leftButton =
      <TouchableHighlight
        underlayColor='silver'
        onPress={ this.props.leftButtonAction }
      >
        <View style={styles.sideButtonsViewWrapper}>
          <Text style={styles.navButtonText}>{this.props.leftButtonText}</Text>
        </View>
      </TouchableHighlight>
    } else if (this.canGoBack()) {
      leftButton =
        <TouchableHighlight
          underlayColor='silver'
          onPress={ this.props.navigator.pop }
        >
          <View style={styles.sideButtonsViewWrapper}>
            <Text style={styles.navButtonText}>Back</Text>
          </View>
        </TouchableHighlight>
    } else {
      {/* This is a bit of a hack,
        for layout to render correctly */}
      leftButton =
        <TouchableHighlight>
          <View style={styles.sideButtonsViewWrapper}/>
        </TouchableHighlight>
    }

    var centerTitle =
      <View style={styles.centerTitleViewWrapper}>
        <Text style={[styles.navButtonText, { fontSize: 22 }]}>
          {this.props.title}
        </Text>
      </View>

    var rightButton =
      <TouchableHighlight
       underlayColor='silver'
       onPress={this.props.rightButtonAction}
      >
        <View style={styles.sideButtonsViewWrapper}>
          <Text style={styles.navButtonText}>{this.props.rightButtonText}</Text>
        </View>
      </TouchableHighlight>

    return (
      <View>
        <SystemBar/>
        <View style={[styles.container]}>
          {leftButton}
          {centerTitle}
          {rightButton}
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
      flex: 1,
      flexDirection: 'row',
      justifyContent: 'space-between',
      height: Config.metrics.navBarHeight,
      backgroundColor: '#1886BC',
  },
  titleBarBackground: {
      backgroundColor: '#cccccc',
      height: Platform.OS === 'ios' ? 20 : 0
  },
  navButtonViewWrapper: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    margin: 10
  },
  navButtonText: {
    fontSize: 16,
    color: 'white'
  },
  centerTitleViewWrapper: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    margin: 10,
  },
  sideButtonsViewWrapper: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    margin: 10,
    width: 50
  }
})