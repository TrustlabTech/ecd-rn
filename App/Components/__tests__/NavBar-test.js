import 'react-native'
import React from 'react'
import NavBar from '../NavBar'

import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(
    <NavBar
      navigator={ {} }
      leftButtonText="Back"
      leftButtonAction={ () => ({}) }
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})