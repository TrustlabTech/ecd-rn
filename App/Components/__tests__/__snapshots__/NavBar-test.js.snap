exports[`test renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "height": 20,
      }
    } />
  <View
    style={
      Object {
        "backgroundColor": "#393939",
        "flex": 1,
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "space-between",
      }
    }>
    <TouchableHighlight
      activeOpacity={0.9}
      onPress={[Function]}
      underlayColor="#dfdfdf">
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#393939",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "width": 60,
          }
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fefefe",
              "fontSize": 16,
            }
          }>
          Back
        </Text>
      </View>
    </TouchableHighlight>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "margin": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fefefe",
              "fontSize": 16,
            },
            Object {
              "fontSize": 22,
            },
          ]
        } />
    </View>
    <TouchableHighlight
      activeOpacity={0.85}
      underlayColor="black">
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#393939",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "width": 60,
          }
        } />
    </TouchableHighlight>
  </View>
</View>
`;
